task2.py

```diff
@@ -0,0 +1,48 @@
+import abc
+
+
+class Shape(object):
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def calc_perimeter(self):
+        """Method documentation"""
+        return
+
+
+class Triangle(Shape):
+
+    def __init__(self, a, b, c):
+        self.a = a
+        self.b = b
+        self.c = c
+
+    def calc_perimeter(self):
+        perim = self.a + self.b + self.c
+        print("Triangle: Consider me implemented", perim)
+        return perim
+
+
+class Rectangle(Shape):
+
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+    def calc_perimeter(self):
+        perim = (self.a + self.b) * 2
+        print("Rectangle perimeter is:", perim)
+        return perim
+
+
+class Square(Rectangle):
+
+    def __init__(self, a):
+        super().__init__(a, a)

```

Comment:
2 Marks - For not having its own calc_perimeter method. As per the diagram, square should use Rectangle’s implementation
2 Marks - For initialising using Rectangle by passing the ‘a’ value twice